features/filesystem/storagelite/source/StorageLite.cpp

```diff
@@ -0,0 +1,1846 @@
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// ----------------------------------------------------------- Includes -----------------------------------------------------------
+
+#include "StorageLite.h"
+
+#if STORAGELITE_ENABLED
+
+#include "aes.h"
+#include "cmac.h"
+#include "sha256.h"
+#include "entropy.h"
+#include "DeviceKey.h"
+#include "BufferedBlockDevice.h"
+#include "nvstore.h"
+#include "mbed_critical.h"
+#include "mbed_assert.h"
+#include "mbed_wait_api.h"
+#include <algorithm>
+#include <string.h>
+#include <stdio.h>
+
+// --------------------------------------------------------- Definitions ----------------------------------------------------------
+
+// Record flags are part of the data_size_and_flags field (both in RAM table and record header)
+// External flags - flags given by the user in set operation
+// Internal flags - flags added by us (like deleted flag)
+static const uint32_t all_rec_flags_mask                    = 0xFFF00000UL;
+static const uint32_t ext_rec_flags_mask                    = 0xFF000000UL;
+static const uint32_t rec_data_size_mask                    = 0x000FFFFFUL;
+// External flags
+const uint32_t StorageLite::rollback_protect_flag           = 0x80000000UL;
+const uint32_t StorageLite::encrypt_flag                    = 0x40000000UL;
+const uint32_t StorageLite::update_backup_flag              = 0x20000000UL;
+// Internal flags
+static const uint32_t delete_flag                           = 0x00100000UL;
+
+// RAM flags are part of the RAM table entry. We use the low bits of the offset for flags,
+// as offset needs to be aligned to 16 bytes (leaving us room for 4 flags)
+static const uint32_t ram_delete_flag                       = 0x00000001UL;
+static const uint32_t ram_is_backup_flag                    = 0x00000002UL;
+static const uint32_t ram_has_backup_flag                   = 0x00000004UL;
+static const uint32_t ram_has_backup_rb_flag                = 0x00000008UL;
+static const uint32_t ram_flags_mask                        = 0x0000000FUL;
+static const uint32_t ram_offset_mask                       = 0xFFFFFFF0UL;
+
+
+static const size_t nonce_size       = 8;
+static const size_t cmac_size        = 16;
+static const size_t full_sha256_size = 32;
+
+typedef struct {
+    uint32_t data_size_and_flags;
+    uint16_t name_size;
+    uint16_t nvstore_key;
+    uint8_t  nonce[nonce_size];
+    uint8_t  cmac[cmac_size];
+} record_header_t;
+
+typedef struct {
+    uint32_t hash;
+    uint32_t offset_and_flags;
+} ram_table_entry_t;
+
+static const uint32_t min_prog_align_size = 16;
+
+static const unsigned char *enc_key_salt  = (const unsigned char *) "Encrypt key salt";
+static const unsigned char *auth_key_salt = (const unsigned char *) "Auth. key salt  ";
```

Comment:
Don't think so. Why lay this burden n the user if used for internal purposes? But I'll change the salt to include "StorageLite", so it won't conflict with other user salt strings.

Reply:
If its a config option, users wouldn't _need_ to know about it. But it would still be available for an application that wants the additional security.

Unsure though, I'll leave it up to you.

Reply:
Kept as is, just changed the salt names to include StorageLite.